core/state_processor.go

```diff
@@ -1175,12 +1215,12 @@ func ProcessQiTx(tx *types.Transaction, chain ChainContext, checkSig, isFirstQiT
 		}
 		ETXPCount++
 		if ETXPCount > *etxPLimit {
-			return nil, nil, fmt.Errorf("tx [%v] emits too many cross-prime ETXs for block. emitted: %d, limit: %d", tx.Hash().Hex(), ETXPCount, etxPLimit), nil
+			return nil, nil, nil, fmt.Errorf("tx [%v] emits too many cross-prime ETXs for block. emitted: %d, limit: %d", tx.Hash().Hex(), ETXPCount, etxPLimit), nil
 		}
-		etxInner := types.ExternalTx{Value: totalConvertQitOut, To: &convertAddress, Sender: common.ZeroAddress(location), OriginatingTxHash: tx.Hash(), Gas: remainingGas.Uint64()} // Value is in Qits not Denomination
+		etxInner := types.ExternalTx{Value: totalConvertQitOut, To: &convertAddress, Sender: common.ZeroAddress(location), EtxType: types.ConversionType, OriginatingTxHash: tx.Hash(), Gas: remainingGas.Uint64()} // Value is in Qits not Denomination
 		*usedGas += params.ETXGas
 		if err := gp.SubGas(params.ETXGas); err != nil {
-			return nil, nil, err, nil
+			return nil, nil, nil, err, nil
 		}
 		etxs = append(etxs, &etxInner)
 		txFeeInQit.Sub(txFeeInQit, txFeeInQit) // Fee goes entirely to gas to pay for conversion
```

Comment:
We will have to rethink this, because I'm not sure conversion txs would be included anymore if there's no miner tip

Reply:
I agree, if the miner starts enforcing the miner tip, we cant take all the fee as gas, but lets think about this, i think its not a blocker to merge this PR